### DISCUSSION TITLE
Add FastFlow, a new Anomaly Detector

### DISCUSSION THREAD

**Discussion by innat** [open-edge-platform/anomalib#189] [Feature Requests] [answered]
**Is your feature request related to a problem? Please describe.**

  - New sota: https://paperswithcode.com/sota/anomaly-detection-on-mvtec-ad
  - paper: https://arxiv.org/pdf/2111.07677v2.pdf

**Comment by bmabir17-asj** [comment]
- unofficial implementation (WIP) : https://github.com/AlessioGalluccio/FastFlow

**Comment by samet-akcay** [comment]
We already have a [FastFlow branch](https://github.com/openvinotoolkit/anomalib/tree/feature/fastflow/anomalib/models/fastflow) to implement it, but it's still in WiP. 

We welcome contributions to the branch. Those who would like to contribute could create a new branch from `feature/fastflow` and create a pull request to this branch.

```
git clone https://github.com/openvinotoolkit/anomalib.git
git checkout feature/fastflow
git checkout -b <name-of-your-branch-to-work-on>
...
# Work on your implementation/changes
...
# Create a PR to feature/fastflow
```

Thanks!

**Comment by innat** [comment]
@samet-akcay Thanks for this information. We're currently working on fastflow, would love to contribute when it's ready.

**Comment by mjack3** [comment]
Did you any progress? I opened my fastflow implementation in my github (wip). The paper is very obscure

**Comment by samet-akcay** [comment]
Not yet. Unfortunately, we've been packed with other tasks that we couldn't find time to work on this

**Comment by mjack3** [comment]
Currently I get -0.5 Image AUROC and -0.6 Pixel AUROC. Some results are pretty good

![image](https://user-images.githubusercontent.com/37028633/162237438-64c60774-d55d-471a-8f6f-3e7e4889d148.png)

but other not

![image](https://user-images.githubusercontent.com/37028633/162237496-13677f88-d268-4db5-bfa4-d674aa9a5dd0.png)
![image](https://user-images.githubusercontent.com/37028633/162237522-acff716e-a956-44aa-8c52-09f4b9c5f7f3.png)

I'm still working on it. This is by using AllInOne block because is the only way to match the trainable parameters or get good results following what paper says. Also note that there is some nonsense in the paper that makes me wonder about its veracity.

**Reply by samet-akcay** [reply]
Thanks for the upate @mjack3. Unfortunately, we don't have much of a progress neither. We could perhaps contact the authors to get some advice?

**Reply by mjack3** [reply]
@samet-akcay that would be great. As far as I know @AlessioGalluccio sended an email to the authors one but I don't have news. [LINK](https://github.com/AlessioGalluccio/FastFlow/issues/14#issuecomment-1036445338)


1- Do you match the trainable parameters for resnet18? I match 2.6M (2.7M paper) using 3-1 convs and 4.7M (4.9 paper) using 3-3.
2- What performance do you achieve?

**Comment by ashwinvaidya17** [comment, marked as answer]
FastFlow was added to Anomalib in this PR https://github.com/openvinotoolkit/anomalib/pull/336
